### DISCUSSION TITLE
Only 5 of 6 plugs work - 6 plug power bar LTS-6A-W5

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#16601] [Q&A] [answered]
https://templates.blakadder.com/ce_smart_home_LTS-6A-W5.html

I flashed this bar and everything but one plug works. Plug #6 will not go on or off.
To determine if the problem is a bad power bar, how could I manually test this using the console, is there a way?

**Comment by barbudor** [comment, marked as answer]
To be sure that it is not a template error, you can try to live the not working Relay_X to other free GPIOs (those defined as None)
Don't try GPIO 9 and 10 if the ESP is 8266, but if it is a 8285 you can also try those (see Information for details on CPU model)

**Reply by dac2020** [reply]
Good to know there's a way but a bit nervous to try this since GPIO are new to me so don't want to ruin this bar.
I'll look around. I'm sure I understand but want to do it right :).

Thanks.

**Reply by dac2020** [reply]
The chip is ESP8266EX.

I found what is happening.
I tried what you said but it doesn't help but then I noticed something using the template in the page above.

If I look at the device with the wire to my left so that the writing is not upside down, then I assume that plug 1 is to the left and plug 6 is to the right.

This image shows the 5th LED on after clicking on button 1.
![plug-021](https://user-images.githubusercontent.com/9727841/191405928-ed3dd1a9-405f-4f5c-8a0e-b7e413264573.jpg)

When I click button 1 on, the 5th plug (counting left to right) actually comes on.
When I click button 2 on, the 4th plug (counting left to right) actually comes on.
When I click button 3 on, the 3rd plug (counting left to right) actually comes on.
When I click button 4 on, the 2nd plug (counting left to right) actually comes on.
When I click button 5 on, the 1st plug (counting left to right) actually comes on.
When I click button 6 on, nothing happens.

![image](https://user-images.githubusercontent.com/9727841/191404675-6d8c3ef3-8ae6-429b-9234-8ef2f2ef93b1.png)

**Reply by barbudor** [reply]
Did you noticed what is expressly written on the strip itself?
The 6th one is not controllable and I believe always on
You are just expecting your hardware to do something it is not designed for

**Reply by dac2020** [reply]
No, I'm a bit blind lol. Wsan't wearing my glasses, just knew the text was not upside down :).

But, did you read what I said?
Even if the 6th one is not meant to work, the order of the buttons don't match the outlets.

Curious why the template has the 6th outlet also?

**Reply by barbudor** [reply]
> Even if the 6th one is not meant to work, the order of the buttons don't match the outlets.

As I don't see any number printed on the outlet itself, it's up to you to have the index sorted in the order you want. Numbering from left to right or right to left is purely your decision.

> Curious why the template has the 6th outlet also?

May be there have been some models which had the 6th socket connected to a relay ?
Manufacturer often change their design without changing the reference

**Reply by dac2020** [reply]
Sure, I can change them for my own use.
I thought someone might want to know so that the outlets could be put into the correct order.
In my mind, outlet 1 is far left.

**Reply by dac2020** [reply]
I guess GPIO are not in order. So far, GPIO2 for example is not related to relay 1.
Reversing 1 and 5 made no difference so I guess I have to learn something new again.

For anyone interested and if you'd like your plugs to start on the left, here are the parameters.

![image](https://user-images.githubusercontent.com/9727841/191584713-2b8f4366-53de-4e58-8869-f150c74fbea9.png)

Blackadder template;
https://templates.blakadder.com/ce_smart_home_LTS-6A-W5.html
{"NAME":"CE Power Strip","GPIO":[288,0,0,0,225,224,0,0,227,226,228,229,32,0],"FLAG":0,"BASE":18}

Mine after update;
{"NAME":"CE Power Strip","GPIO":[288,0,0,0,227,228,0,0,225,226,224,0,32,0],"FLAG":0,"BASE":18}

**Reply by sfromis** [reply]
If you have an improved template making more sense, updates are welcomed by the templates repository.

**Reply by dac2020** [reply]
Not sure I'd call it improved but if someone wants to put their sockets in some order instead of random, they are welcome to use mine :).

**Reply by sfromis** [reply]
I'd say that most users would agree that number 1-5 in the reading direction of the text on the strip makes better sense. And removing the 6th channel surely makes sense if it is not doing anything :-)

**Reply by dac2020** [reply]
If whom ever maintains the templates sees this, feel free to use my changes then.

**Reply by barbudor** [reply]
Like everything in Tasmota, this is contribution based. feel free to suggest your changes there https://github.com/blakadder/templates

**Reply by sfromis** [reply]
Each template page also have a direct "Edit on Github" link to the right, inviting contributions.

**Reply by dac2020** [reply]
Back and I've added a change in the template. Not sure it's how you want it done however.

**Reply by dac2020** [reply]
Can someone confirm if I did it correctly?

**Reply by barbudor** [reply]
As I don't seen any change (Relay 6 is still here), probably not
You seem to have forked and made a change in your fork, branch patch-1
But you haven't submited a PR

**Comment by dac2020** [comment]
I clicked on edit then added my information and saved. It showed it was waiting to be confirmed or something but that's gone now and I only see the original. I can try again later.

**Reply by barbudor** [reply]
You have your change saved here : https://github.com/dac2020/templates/blob/patch-1/_templates/ce_smart_home_LTS-6A-W5
It's a fork of blakadder/templates repository, with an automatically created new branch called "patch-1"
Not you have to make a PR
Go back to the root of the repo in your fork, branch patch-1 : https://github.com/dac2020/templates/tree/patch-1
It says "1 commit ahead" which is your change
Click on the "Contribute" and "Open Pull Request"
Fill-up the description why you are suggesting this change
and click on "Create Pull Request"

**Reply by dac2020** [reply]
Done.

**Comment by dac2020** [comment]
I wanted to flash another of these and get;

Your device's firmware is too new.
Tuya patched the PSK vulnerability that we use to establish a connection.
You might still be able to flash this device over serial.

Since this was mentioned, does it mean no way to do it other than serial or that I need a newer version of tuya-convert?

**Reply by sfromis** [reply]
Tuya-convert does not appear to be maintained any more, unless someone out of the blue manages to find a new hack, which seems to have been too difficult. I know of no other method than wired serial flashing.

**Reply by dac2020** [reply]
I wonder if the person was threatened by tuya or some other company to cease.
Is it even legal to take someone's hardware and convert it from their fw to your own? I really don't know.

**Reply by dac2020** [reply]
Really, I don't understand this game of constantly trying to find hardware that will run tasmota.

Looking on amazon, it's always a hit and miss. 
Looking at the list provided by tasmota, it's always a hit and miss because many have broken links and others you have to take your chances.
Asking sellers on aliexpress leads to their pretending they don't understand English unless you're asking about buying and not about the firmware. 

I hate that because we should have the right to run what ever we want on hardware that we purchase. 
I hate that Google controls most of the Internet in one way or another. 
I hate that software makers pretty much own my PC.

It's too bad more people don't pay attention and take part in projects like this. If I can't find hardware, I'll have to find some other firmware or devices that run mqtt by default.

**Reply by dac2020** [reply]
Does anyone know of a document that shows how to convert these using serial?
I find nothing on the Internet.

**Reply by barbudor** [reply]
Google doesn't want you to flash those devices 😄 

At some point you will have to open it to serial-flash it.
So start by that, find the ESP
Is there an obvious connector or pads that could be labelled like GND, TX, RX, IO0 ?
Is that a known module ?  https://tasmota.github.io/docs/Pinouts/

If you don't like the game's rule, change for another game

**Reply by dac2020** [reply]
I simply don't have the time to do that. I love to push these kinds of projects but I have limited time to play with these things.
Anyhow, I found this so will look.

https://tasmota.github.io/docs/Getting-Started/

**Reply by dac2020** [reply]
It's A TYWE3S chip but I can't solder onto such small leads and have nothing that could clamp around such small leads.

**Reply by dac2020** [reply]
Does anyone make something that could sit on top of the chip, pressed in place touching the leads? 
I'm sure that exists and would make doing this much less painful.

**Reply by sfromis** [reply]
I don't think there was any threat or other pressure. Basically, Tuya simply improved security by fixing flaws in their software. The words you posted does not even imply any discomfort. Likewise I don't think that they were worrying about a few crazies (us) replacing firmware on a minute part of their sales.

About the legality, I have heard of nothing but people taking a piece of hardware **they own**, and removing one part (the firmware) and replacing it. I'd see legal risks if the process was based on reverse engineering or modifying Tuya firmware, or using the the Tuya backend (cloud servers) outside of what would happen with Tuya firmware. This was not the case, the devices simply lost any connection to Tuya servers.

Another legal risk could be users getting into trouble with the device after replacing firmware, and laying liability (legal or reputational damage) on Tuya without them being at fault about the problems. I've not heard of such cases, and I'd expect most people to understand that they basically made the choice of breaking the device by replacing firmware, and cutting any relationship to Tuya.

From the perspective of Tuya, they made a sale and earned a bit of money. Contrary to what some people would have the public believe, data potentially collected (if Tuya even does that) from individual users is not of much value (I did not say absolutely zero, just not much). Arguably, having such data on very large numbers of users can have some value, sort of being market research pointing at habits and possible new product opportunities, but a few defectors (us) does not make a real difference there.

The larger loss, seen from a Tuya perspective, may be that users enlisting as members the Tuya ecosystem would likely be inclined to specifically select further products from that ecosystem, while the few crazies have been free to mix and match without any reason to select Tuya-based products over other ecosystems.

You could even say that having an easy firmware replacement option for Tuya devices made the small nerd segment (us crazies) hold a more favorable opinion of Tuya, and act as "brand ambassadors" recommending the products in their networks. This potential benefit was lost by firmware replacement no longer being as feasible, with users likely going from being positive to being negative about Tuya, and less inclined to recommend the products to less tech-savvy friends and family. But again, my view is that the numbers are just too small for this to be a significant consideration, if Tuya even hypothetically thought about such ideas.

**Reply by sfromis** [reply]
If you want to break out of the game with firmware-locked devices, Zigbee looks to be a good avenue. Devices can be enrolled in a local-only setup without any kind of hardware or firmware modification. Someone may have to work a bit with the Zigbee protocol details to get all details to work, especially as Tuya is a standards-breaker, but it is generally feasible to use Zigbee devices, just with the loss of very detailed on-device control possible with firmware replacement. But most devices are not that interesting for making alternate controls anyway.

**Reply by sfromis** [reply]
Here's a jig for TYWE3S: https://www.youtube.com/watch?v=imKZbhJ8lvU

**Reply by dac2020** [reply]
Something like this too.

https://www.ebay.com/itm/284611317950
and
https://www.ebay.com/itm/134118373508

**Reply by dac2020** [reply]
Alright, I ordered an IC test clip and some micro clips and will try to give this a shot once those come in.

**Reply by sfromis** [reply]
The test clip you linked was for SOP16 which is 1.27 mm pitch, while the ESP-12 form factor modules (including TYWE3S) is 2.0 mm pitch. And that type of clip may not work well for the castellated edge pins of that form factor. Likewise those "tweezer" clamps from your 2nd link may not find much to grip on when the ESP-12 form factor modules have no regular pins, but are SMD-like soldered flat to the main PCB.

**Reply by dac2020** [reply]
I ended up ordering these from Amazon but I can cancel them and get the more correct stuff. This is all new to me so I expect to make mistakes. If you know what I should get, please share as I'm willing to at least try the serial method if it can give me better luck moving forward.

https://www.amazon.com/gp/product/B08SQR7QQJ
https://www.amazon.com/gp/product/B07F67VGJF

**Reply by dac2020** [reply]
I can't find a pitch 2.00mm 16 pin clip anywhere so far.

**Reply by dac2020** [reply]
I ordered a bunch of micro test clips in different configurations hoping those might work.

**Reply by sfromis** [reply]
Hence people are 3D printing jigs with pogo pins....

**Comment by dac2020** [comment]
I thought you said something important but I cannot find it now. I think you said as long as it's an ESP chip, it can be converted to tasmota from serial if nothing else.

**Reply by sfromis** [reply]
Don't know what you are talking about, but with ESP8266 there is no way of blocking flashing. Still some types of hardware design, like having another MCU on serial, can take extra steps to make flashing work. And ESP32 chips does have the feature of fuse bits which can allow the manufacturer to close the door to reflashing via serial.

**Reply by dac2020** [reply]
>Don't know what you are talking about, but with ESP8266 there is no way of blocking flashing.

Then you do know what I'm talking about because that was the question :).
You made a mention of this somewhere. I thought it was in this thread but maybe it was somewhere else.
The point being that if I know I can always flash using serial, then it is better that I learn to do that rather than buying things that are a hit and miss. I'll have a better and second chance of flashing devices. 

All this of course, assuming manufacturers don't get forced into new lays that as is mentioned below, make it harder to flash.

**Reply by sfromis** [reply]
Not hard to do, a bit of soldering can be picked up quickly - sometimes devices can be tricky to open. But as I already mentioned, Zigbee may be a quite viable strategy for local-only control without messing with device internals, and ESP8266 usage is dwindling.

**Reply by dac2020** [reply]
Does tasmota follow some sort of MQTT or other standard? I mean, I've posted asking how to do this and that using tasmota comments but I've no idea if those are specific to tasmota or common.

For example, being able to leave the config but reset only to AP mode. That's pretty useful or being able to add TLS etc.
Yes, I'll look into others but was trying to stick with this project to kind of support it at the same time.

**Reply by sfromis** [reply]
MQTT is by itself a common standard protocol, which Tasmota uses. Payload format has no formal standard, but Tasmota is using a JSON format which should be pretty easy to deal with, both for parsers and humans. Do not expect other projects to use the same formats, configs and commands. My preference is to standardize stuff to use the same project, which has become Tasmota.

**Reply by dac2020** [reply]
That's why I've been trying to stick with tasmota. Thanks for the info.

**Comment by barbudor** [comment]
EU is currently working on a legislation to improve security of IOT and home automation devices
I don't expect the law to be very detailed on what would need to be prevented but surely OTA update of non certified firmware is certainly in the focus
Preventing serial flashing may not be key but might also be covered or at least considered by the manufacturers as a must have to comply

The goal is not against us tinkerers (they don't care of us) but to prevent someone from the outside to take control of someone home. 
You can't expect Mr Random to be a security expert so manufacturers have to provide products that are secured by design

That might make our life more and more difficult

**Reply by sfromis** [reply]
Let's see how tinkerer-friendly EU based Shelly accommodates potential restrictions, with both OTA and wired flashing possible on many current models.

**Comment by dac2020** [comment]
BTW, was my template submission ok?

**Reply by sfromis** [reply]
What I see on the templates repository page looks a lot like your improved template :+1: 
https://templates.blakadder.com/ce_smart_home_LTS-6A-W5.html
